src/mavsdk/plugins/camera/camera_definition.cpp

```diff
@@ -196,6 +199,17 @@ bool CameraDefinition::parse_xml()
             continue;
         }
 
+        auto get_default_opt = [&]() {
+            auto maybe_default = find_default(new_parameter->options, default_str);
+
+            if (!maybe_default.first) {
+                LogWarn() << "Default not found for " << param_name;
+                return std::optional<Option>{};
+            }
+
+            return std::optional{maybe_default.second};
+        };
```

Comment:
Nice!

Reply:
Thx! :) About the std::optional - I noticed that there are some usages of `std::pair<bool` which could be done (semantically) better with `std::optional`. I might give it a shot changing it if you think it's a good idea.

Reply:
Internally, `std::optional` is good. However, in the external (auto-generated) API it's not an option. I think there we only have `std::pair`.

Reply:
Got it, I rather won't touch it then 👍.